src/Lean/Meta/Closure.lean

```diff
@@ -398,6 +398,9 @@ def mkAuxDefinitionFor (name : Name) (value : Expr) (zetaDelta : Bool := false)
 -/
 def mkAuxTheorem (type : Expr) (value : Expr) (zetaDelta : Bool := false) (kind? : Option Name := none) (cache := true) : MetaM Expr := do
   let result ← Closure.mkValueTypeClosure type value zetaDelta
+  if result.value.hasFVar then
+    pure <| panic! "Closure.mkValueTypeClosure left a stray fvar; skipping `mkAuxTheorem` and returning the expression unmodified"
```

Comment:
This doesn't actually work, c.f. [#lean4 > Non-fatal panics in IO @ 💬](https://leanprover.zulipchat.com/#narrow/channel/270676-lean4/topic/Non-fatal.20panics.20in.20IO/near/543609195)

Reply:
```suggestion
    -- enter BaseIO so that the panic logs rather than throws
    show BaseIO Unit from panic! "Closure.mkValueTypeClosure left a stray fvar; skipping `mkAuxTheorem` and returning the expression unmodified"
```